kernel/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -355,74 +346,150 @@ private Module resolveConfigBubbles(Definition def, Module inputModule, RuleGram
                 throw KEMException.compilerError("Module Map must be visible at the configuration declaration, in module " + module.name());
             }
             return Module(module.name(), (Set<Module>) module.imports().$bar(Set(mapModule)),
-                    (Set<Sentence>) module.localSentences().$bar(configDeclRules),
+                    (Set<Sentence>) module.localSentences().$bar(configDeclRules)
+                            .filter(s -> !(s instanceof Bubble && ((Bubble) s).sentenceType().equals(configuration))),
                     module.att());
         }
 
     }
 
-    private Module resolveNonConfigBubbles(Module module, Scanner scanner, RuleGrammarGenerator gen) {
-        if (stream(module.localSentences())
-                .filter(s -> s instanceof Bubble)
-                .map(b -> (Bubble) b)
-                .filter(b -> !b.sentenceType().equals(configuration)).count() == 0)
-            return module;
-
-        Module ruleParserModule = gen.getRuleGrammar(module);
+    // Find the entry modules (not included in any other module)
+    private static java.util.Set<Module> getBotModules(Set<Module> allModules) {
+        java.util.Set<Module> included = new HashSet<>();
+        for (Module m : mutable(allModules)) {
+            included.addAll(mutable(m.importedModules()));
+        }
+        java.util.Set<Module> rez = mutable(allModules);
+        rez.removeAll(included);
+        return rez;
+    }
```

Comment:
Should we call this `getEntryModules` instead, since that's the terminology we use everywhere else?

Reply:
`Definition.entryModules` is where we store the modules visible by the user.
Other modules not visible by the user include helper modules for parsing (*-CONFIG-CELLS), the syntax split (*$SYNTAX). I guess my comment is not that intuitive then. I will change it.

Reply:
I guess I would call them `topModules` then, not `botModules`. They are at the top of the import chains.